python/lsst/ts/vent/controller/dispatcher.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import traceback
+
+from lsst.ts import tcpip
+
+from .controller import Controller
+
+
+def _type_convert(t: type, value: str):
+    """Converts the value string to the specified type. In the case of boolean,
+    "True" or "T" or "1" is ``True`` and everything else is ``False``. Other
+    cases are handled by cast.
+
+    Parameters
+    ----------
+    t: type
+        The type that value should be converted to.
+
+    value: str
+        The value to be converted.
+
+    Raises
+    ------
+    ValueError
+        If the value cannot be converted to the specified type.
+    """
+
+    if t == bool:  # Boolean is a special case
+        return value.lower() in ("true", "t", "1")
+    return t(value)
+
+
+class Dispatcher(tcpip.OneClientReadLoopServer):
+    def __init__(
+        self, port: int, log: logging.Logger, controller: Controller = Controller()
+    ):
+        self.dispatch_dict = {
+            "closeVentGate": [int],
+            "openVentGate": [int],
+            "resetExtractionFanDrive": [],
+            "setExtractionFanDriveFreq": [float],
+            "setExtractionFanManualControlMode": [bool],
+            "startExtractionFan": [],
+            "stopExtractionFan": [],
+            "ping": [],
+        }
+        self.controller = controller
+
+        super().__init__(port=port, log=log, terminator=b"\r")
+
+    async def respond(self, message: str) -> None:
+        await self.write_str(message + "\r\n")
+
+    async def read_and_dispatch(self) -> None:
+        """Read, parse and execute a command, and send a response."""
+        data = await self.read_str()
+        data = data.strip()
+        if not data:
+            return
+
+        self.log.debug(f"Received command: {data!r}")
+
+        command, *args = data.split()
+        if command not in self.dispatch_dict:
+            await self.respond(f"{command} raise NotImplementedError()")
```

Comment:
uhm.. this is a bit weird, what is this supposed to do?

Reply:
Line-by-line comments added that I hope will clarify what I'm trying to do here.